### PR TITLE
Fix/adm 937 [frontend]

### PR DESCRIPTION
**Pull request by zhou-yinyuan**
## Summary

Pipeline crew setting could filter out dependentbot pipeline

## Before

1. When pipeline crews exist in config file, and a part of crews don't exist in github author, it will still show in pipeline crews selected input.

image

In the picture, the heartbeat-user is from config file but we don't get it from backend.

2. When we get pipeline crews, we get many data that is repeated.

image

In the picture, the `TingyuDong` and `Tingyu Dong` should be the same one.

## After

1. We only show the selected pipeline crews in config file and backend data at the first time.

image

in the picture, we find the heartbeat-user doesn't show in input at the first time.

2. We use author.username to replace author.name

image

In the picture, `TingyuDong` appears only once

### CODE REVIEW

File: frontend/__tests__/containers/MetricsStep/Crews.test.tsx [hunk 1]
```diff
@@ -47,17 +42,29 @@ describe('Crew', () => {
   });
 
   it('should selected all options by default when initializing given type is board', () => {
+    store.dispatch(saveUsers(['crew A', 'crew B']));
```
Comment (guzhongren) [inline frontend/__tests__/containers/MetricsStep/Crews.test.tsx:45]: duplicated code

File: frontend/__tests__/containers/MetricsStep/Crews.test.tsx [hunk 2]
```diff
@@ -47,17 +42,29 @@ describe('Crew', () => {
...
+  it('should show detail options when initializing given type is other and click Included crews button', async () => {
+    store.dispatch(savePipelineCrews(['crew B', 'crew C']));
     setup('other');
 
-    expect(screen.getByRole('button', { name: 'A' })).toBeInTheDocument();
-    expect(screen.getByRole('button', { name: 'B' })).toBeInTheDocument();
+    await act(async () => {
+      await userEvent.click(screen.getByRole('combobox', { name: mockLabel }));
+    });
+    const listBox = within(screen.getByRole('listbox'));
```
Comment (guzhongren) [inline frontend/__tests__/containers/MetricsStep/Crews.test.tsx:59]: new line before this line

**Review by guzhongren — APPROVED** [review]

File: frontend/__tests__/containers/MetricsStep/Crews.test.tsx (hunk 1 above)
Comment (zhou-yinyuan) [reply frontend/__tests__/containers/MetricsStep/Crews.test.tsx]: I moved this code to beforeEach

File: frontend/__tests__/containers/MetricsStep/Crews.test.tsx (hunk 2 above)
Comment (zhou-yinyuan) [reply frontend/__tests__/containers/MetricsStep/Crews.test.tsx:59]: done

File: frontend/__tests__/containers/MetricsStep/Crews.test.tsx (hunk 1 above)
Comment (zhou-yinyuan) [reply frontend/__tests__/containers/MetricsStep/Crews.test.tsx]: done

**Review by weiraneve — APPROVED** [review]
LGTM

File: backend/src/main/java/heartbeat/service/report/PipelineService.java [hunk 3]
```diff
@@ -149,9 +149,11 @@ private List<BuildKiteBuildInfo> getBuildKiteBuildInfo(String startTime, String
 				startTime, endTime);
 
 		if (!CollectionUtils.isEmpty(pipelineCrews)) {
+			buildKiteBuildInfo.stream()
+				.filter(it -> it.getAuthor() == null)
+				.forEach(it -> it.setAuthor(BuildKiteBuildInfo.Author.builder().username("Unknown").build()));
 			buildKiteBuildInfo = buildKiteBuildInfo.stream()
-				.filter(info -> ((pipelineCrews.contains("Unknown") && info.getAuthor() == null))
-						|| (info.getAuthor() != null && pipelineCrews.contains(info.getAuthor().getName())))
+				.filter(info -> pipelineCrews.contains(info.getAuthor().getUsername()))
```
Comment (mrcuriosity-tw) [inline backend/src/main/java/heartbeat/service/report/PipelineService.java:156]: I didn't get this intention of refactor, let's talk about it in CR.

**Review by mrcuriosity-tw — COMMENTED** [review]
Another question, I saw a lot of download data change in e2e/fixtures, but didn't find the time-range fixture change.
Does that mean the pipeline crews calculation logic affects the report data calculation?

**Review by PengxiWPix — APPROVED** [review]
LGTM

**Review by guzhongren — APPROVED** [review]

**Pull request merged**